server.c

```diff
@@ -5258,7 +5315,11 @@ handle_tcp_accept(int fd, short event, void* arg)
 	tcp_data->query_count = 0;
 #ifdef HAVE_SSL
 	tcp_data->shake_state = tls_hs_none;
-	tcp_data->tls = NULL;
+	/* initialize both incase of dangling pointers */
+	if (data->tls_accept || data->tls_auth_accept) {
+		tcp_data->tls = NULL;
+		tcp_data->tls_auth = NULL;
+	}
```

Comment:
I think it could always initialize the member variables to NULL? That may be more prudent. Is it correct that this is possible? If so the code suggestion:
```suggestion
	/* initialize both incase of dangling pointers */
	tcp_data->tls = NULL;
	tcp_data->tls_auth = NULL;
```

Reply:
This is tricky and we need to be carefull here.
I has a hude bug here and random crashes untill I did null but tls_auth and tls.

I had cases where I was in tls and tls_auth was not null, that's why I initialized both of them.
I believe your change is fine, its a more general rule. To my understadning handle_tcp_accept() is unique for each connection.

Reply:
We leave this as it is?
I believe it does not hurt to initialize every time, apart from the extra cycles when we're not in a TLS listener.

Reply:
The commit already changes the code https://github.com/NLnetLabs/nsd/pull/337/commits/a19841c18a50c8368ad8aaf326fcc6ee785f1852 . So I think it is fine, I wanted to make sure it was okay, which edit is the one you think best here.